cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -75,6 +74,43 @@ public Profile getProfile(ProfileId profileId) throws NotFoundException {
     }, NotFoundException.class);
   }
 
+  /**
+   * Get the profile information about the given profile with property overrides. If a non-editable property is
+   * in the overrides, it will be ignored.
+   *
+   * @param profileId the id of the profile to look up
+   * @param overrides overrides to the profile properties
+   * @return the profile information about the given profile
+   * @throws NotFoundException if the profile is not found
+   */
+  public Profile getProfile(ProfileId profileId, Map<String, String> overrides) throws NotFoundException {
+    Profile storedProfile = getProfile(profileId);
+
+    List<ProvisionerPropertyValue> properties = new ArrayList<>();
+    Set<String> remainingOverrides = new HashSet<>(overrides.keySet());
+
+    // add all  properties from the stored profile
+    for (ProvisionerPropertyValue storedProperty : storedProfile.getProvisioner().getProperties()) {
+      String propertyName = storedProperty.getName();
+      String storedVal = storedProperty.getValue();
+      if (!storedProperty.isEditable()) {
+        properties.add(storedProperty);
```

Comment:
Like mentioned above, should we log a message here about the property not get overriden?

Reply:
yeah i'll log something about it.